data/random-teams.ts

```diff
@@ -497,48 +549,267 @@ export class RandomTeams {
 		return nPokemon;
 	}
 
+	customRandomNPokemon(ruleTable: RuleTable, n: number, requiredType?: string) {
```

Comment:
There's a lot of repeated code here - why make a separate function rather than just adding an optional `ruleTable` parameter to `randomNPokemon` and thinning the pool based on that if it's specified?

Reply:
I wasn't sure how much that was going to diverge, but I think you're right, it's unnecessary to make it a separate method now, so I merged them together. I also wasn't sure if the minSourceGen parameter was still necessary: I have kept it in the merged randomNPokemon but it doesn't seem to be referenced anywhere.

Reply:
I think I might've used in an implementation of a custom rule which became infeasible for other reasons? It's not used in private code, though; if it's not referenced anywhere it should be safe to remove.